src/useFilterSubscribe.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useEffect, useState } from "react";
+import type { IDecodedMessage, IDecoder, Waku } from "@waku/interfaces";
+
+import type { HookState } from "./types";
+
+type UseFilterSubscribeParams = {
+  waku: Waku;
+  decoder: IDecoder<IDecodedMessage>;
+};
+
+type UseFilterSubscribeResult = HookState & {
+  messages: IDecodedMessage[];
+};
+
+export const useFilterSubscribe = (
+  params: UseFilterSubscribeParams,
+): UseFilterSubscribeResult => {
+  const { waku, decoder } = params;
+
+  const [error, setError] = useState<null | string>(null);
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const [messages, setMessage] = useState<IDecodedMessage[]>([]);
+
+  const pushMessage = useCallback(
+    (message: IDecodedMessage): void => {
+      setMessage((prev) => [...prev, message]);
+    },
+    [setMessage],
+  );
+
+  useEffect(() => {
+    let unsubscribe: null | (() => Promise<void>) = null;
+    setLoading(true);
+
+    waku?.filter
```

Comment:
An idea I want to convey (no immediate action needed). Is that currently, you need to provide a hook for filter and relay to receive messages.
And the developer need to use the right hook depending on whether they created a light node or relay node.


With the introduction of the `Sender` and `Receiver` interfaces to improve the noise API. I hope we can have the Light node and relay node implement an interface like that:


```
interface WakuNode {
   sender: Sender;
   receiver: Receiver
}
```

So then, when you want to receive message, you always do:

```
node.receiver.addObserver(...)
```

Whether you use relay or filter.

Cc @danisharora099 FYI for this proposed interface.